contracts/BondToken.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+import "./HubPoolInterface.sol";
+import "./WETH9.sol";
+
+interface _HubPool is HubPoolInterface {
+    // Specify the automatically-implemented rootBundleProposer() getter.
+    function rootBundleProposal() external pure returns (HubPoolInterface.RootBundle memory);
+}
+
+/**
+ * @notice Across Bond Token (ABT).
+ * ABT is a simple deposit contract based on WETH9. ABT is issued proportionally to any address that deposits Ether. It
+ * imposes address-based permissioning on the WETH9 transferFrom() function in order to constrain the movement of ABT
+ * into the Across v2 HubPool contract. When configured as the required HubPool bond token, ABT can dramatically reduce
+ * the attack surface of the HubPool by requiring that addresses are explicitly approved before they can successfully
+ * submit a root bundle proposal. The address-based permissiong does not constrain transfers that are needed to dispute
+ * a root bundle proposal, so the ability of decentralised/unknown actors to dispute is unaffected.
+ */
+contract BondToken is WETH9, Ownable {
+    using Address for address;
+
+    _HubPool public immutable hubPool;
+
+    /**
+     * @notice Addresses that are permitted to make HubPool root bundle proposals.
+     */
+    mapping(address => bool) public proposers;
+
+    /**
+     * @notice Emitted on proposer permissions update.
+     */
+    event ProposerModified(address proposer, bool enabled);
+
+    /**
+     * @notice BondToken constructor.
+     * @param _hubPool Address of the target HubPool contract.
+     */
+    constructor(_HubPool _hubPool) {
+        name = "Across v2 Bond Token";
```

Comment:
```suggestion
        name = "Across Bond Token";
```
Not sure V2 is necessary. Do you think we should name it Proposer Bond instead of just Bond?

Reply:
Preface: Naming things is hard :)

My thinking with the "v2" naming is that this token has a finite lifespan - it will only be in use as long as the current (Across v2) HubPool is deployed. That's why I preferred to include v2 in the name.

Reply:
I generally don't like Versions in token names, look at UMA Voting Token V1 for example

Reply:
"v2" removed: https://github.com/across-protocol/contracts-v2/pull/226/commits/ee0aedb78f29999d9518bec709beeaa13749e9a4

I do think it's a different scenario to UMA v1, but I can live with either :)